### DISCUSSION TITLE
请教升级到最新版本之后，很多模块判定为不是esm的类型，于是会持续被重复加载

### DISCUSSION THREAD

**Discussion by watsonsong** [Tencent/puerts#1571] [Q&A]
升级之后产生一个问题，在import或者require的时候，如果两个ts文件有相互import的关系，然后产生了`JavaScript\puerts\modular.js:63: RangeError: Maximum call stack size exceeded`的错误。
回退回1.0.5版本的modular文件则一切正常。看起来是在let m = {exports: {}};之后，并没有提前保存到缓存。
```
localModuleCache[moduleName] = m;
moduleCache[key] = m;
```

这样executeModule内部再发生require的时候会导致重复递归

**Comment by chexiongsheng** [comment]
互相import在nodejs那也可能触发错误。本来循环依赖就应该业务避免。
之前可以是因为之前有个bug：https://github.com/Tencent/puerts/commit/24b1b37e416482823bb52297be900d81a117f862
这会导致require出错后，第二次require反而正常了。

**Reply by watsonsong** [reply]
是这个改动产生的，但感觉应该先于executeModule设置进去，出错了再清理。
否则executeModule内部如果递归到了自己，还会判断自己没有在cache中，导致重复多次加载。

**Reply by chexiongsheng** [reply]
这种改法也可以解决问题，你改下，然后PR给我。

**Reply by watsonsong** [reply]
做了一个尝试改动，似乎没有引起其他问题，看看这么调整一下顺序是否合适？
https://github.com/Tencent/puerts/pull/1583
